src/remote-backend/domain-id.cc

```diff
@@ -0,0 +1,31 @@
+#include "remote-backend.hh"
+
+namespace pdns_sysrepo::remote_backend
+{
+uint32_t getNewId(const boost::bimap<std::string, uint32_t> &map) {
+  if (map.empty()) {
+    return 1;
+  }
+  uint32_t highest = map.left.rbegin()->second;
```

Comment:
Shouldn't this be `map.right.rbegin()->first`?

Reply:
This works in the tests. The map.left is <string, uint32_t>.